src/api/app/services/monitor_controller_service/building_information_updater.rb

```diff
@@ -0,0 +1,50 @@
+module MonitorControllerService
+  class BuildingInformationUpdater
+    attr_accessor :workers
+
+    def initialize
+      @workers = {}
+    end
+
+    def call
+      initialize_workers
+      update_workers
+      self
+    end
+
+    private
+
+    def worker_status
+      @worker_status ||= Xmlhash.parse(WorkerStatus.hidden.to_xml)
+    end
+
+    def initialize_workers
+      @workers = Hash[worker_status.elements('idle').collect { |b| [worker_id(b), {}] }]
+    end
+
+    def update_workers
+      @workers.merge!(Hash[@worker_status.elements('building').collect { |b| [worker_id(b), workers_hash(b, calculate_delta(b['starttime'].to_i))] }])
+    end
+
+    def workers_hash(b, delta)
+      Hash[['project', 'repository', 'package', 'arch', 'starttime'].collect { |k| [k, b[k]] }].merge('delta' => delta)
```

Comment:
Looks like this is filtering some keys and discarding the rest, then merging the `delta`. If so, you can use `Hash#slice` to filter `b` and then `merge`

Reply:
I don't see any advantage on that. Do you want to benchmark it?

Reply:
No, I don't want to benchmark it, its just that to me it feels easier to read like this:
```
def workers_hash(b, delta)
  b.slice('project', 'repository', 'package', 'arch', 'starttime').merge('delta' => delta)
end
```

Reply:
great @danidoni, updated, thanks!